test/ecto/changeset_test.exs

```diff
@@ -50,8 +50,8 @@ defmodule Ecto.ChangesetTest do
       field :source, :map
       field :permalink, :string, source: :url
       belongs_to :category, Ecto.ChangesetTest.Category, source: :cat_id
-      has_many :comments, Ecto.ChangesetTest.Comment, on_replace: :delete
-      has_one :comment, Ecto.ChangesetTest.Comment
+      has_many :comments, Ecto.ChangesetTest.Comment, on_replace: :delete, references: :id
```

Comment:
This is the scenario mentioned in the PR description where one of multiple primary keys is a assumed and maybe shouldn't be.

The two primary keys
1) hidden via the default `@primary_key {:id, :id, autogenerate: true}`
2) defined by the field call `field :token, :integer, primary_key: true`

Current implementation only looks at `@primary_key` and ignores the field-defined primary key